#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/Floor.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"

WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.24764799111930622 -0.4431853897342645 -0.8615434886415544 1.7962449557525686
  position 0.05330155668328318 4.253379227820017 -0.7433230244582903
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF Floor Floor {
  rotation 1 0 0 -1.5708003061004252
  size 4 4
  tileSize 0.4 0.4
}
DEF Walls Group {
  children [
    DEF Wall Solid {
      translation -0.9898 0.075 0.4242
      rotation 0 1 0 -0.7853996938995746
      children [
        DEF Wall Shape {
          appearance PBRAppearance {
            baseColor 0.392157 0.392157 0.392157
            roughness 1
            metalness 0
          }
          geometry Box {
            size 1.2 0.15 0.015
          }
        }
      ]
      name "solid(9)"
      boundingObject USE Wall
    }
    DEF Wall Solid {
      translation -0.148347 0.075 -0.417253
      rotation 0 1 0 -0.7853996938995746
      children [
        DEF Wall Shape {
          appearance PBRAppearance {
            baseColor 0.392157 0.392157 0.392157
            roughness 1
            metalness 0
          }
          geometry Box {
            size 1.2 0.15 0.015
          }
        }
      ]
      name "solid(2)"
      boundingObject USE Wall
    }
    DEF Wall Solid {
      translation -0.9898 0.075 -0.4242
      rotation 0 1 0 0.7853996938995746
      children [
        DEF Wall Shape {
          appearance PBRAppearance {
            baseColor 0.392157 0.392157 0.392157
            roughness 1
            metalness 0
          }
          geometry Box {
            size 1.2 0.15 0.015
          }
        }
      ]
      name "solid(8)"
      boundingObject USE Wall
    }
    DEF Wall Solid {
      translation -0.148348 0.075 0.417253
      rotation 0 1 0 0.7853996938995746
      children [
        DEF Wall Shape {
          appearance PBRAppearance {
            baseColor 0.392157 0.392157 0.392157
            roughness 1
            metalness 0
          }
          geometry Box {
            size 1.2 0.15 0.015
          }
        }
      ]
      name "solid(3)"
      boundingObject USE Wall
    }
    DEF Wall Solid {
      translation -0.841349 0.075 -0.275749
      rotation 0 1 0 0.7853996938995746
      children [
        DEF Wall Shape {
          appearance PBRAppearance {
            baseColor 0.392157 0.392157 0.392157
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.8 0.15 0.015
          }
        }
      ]
      name "solid(13)"
      boundingObject USE Wall
    }
    DEF Wall Solid {
      translation -0.282739 0.075 0.282862
      rotation 0 1 0 0.7853996938995746
      children [
        DEF Wall Shape {
          appearance PBRAppearance {
            baseColor 0.392157 0.392157 0.392157
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.8 0.15 0.015
          }
        }
      ]
      name "solid(1)"
      boundingObject USE Wall
    }
    DEF Wall Solid {
      translation -0.84842 0.075 0.28282
      rotation 0 1 0 2.3561996938995744
      children [
        DEF Wall Shape {
          appearance PBRAppearance {
            baseColor 0.392157 0.392157 0.392157
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.8 0.15 0.015
          }
        }
      ]
      name "solid(12)"
      boundingObject USE Wall
    }
    DEF Wall Solid {
      translation -0.289811 0.075 -0.275792
      rotation 0 1 0 2.3561996938995744
      children [
        DEF Wall Shape {
          appearance PBRAppearance {
            baseColor 0.392157 0.392157 0.392157
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.8 0.15 0.015
          }
        }
      ]
      name "solid(11)"
      boundingObject USE Wall
    }
  ]
}
DEF ROBOT_RED Robot {
  translation -0.572739 0 -0.714217
  rotation 0 1 0 2.3561996938995744
  children [
    Gyro {
    }
    Accelerometer {
    }
    Lidar {
      rotation -0.5773502691896258 0.5773502691896258 0.5773502691896258 2.0943951023931957
    }
    DEF MAIN_BODY Pose {
      translation 0 0.0415 0
      rotation -1 0 0 1.5707963267948966
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.196078 0.196078 0.196078
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    DEF LEFT_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        position 716.4543820994643
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
        }
      ]
      endPoint DEF LEFT_WHEEL Solid {
        translation -0.02999999999999456 0.025 0
        rotation 0.9999999999999999 0 0 4.813412919008605
        children [
          DEF WHEEL Pose {
            rotation -0.5776567755050862 -0.5771969549958951 0.5771969549958951 2.0939354039397986
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.196078 0.196078 0.196078
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject DEF WHEEL Pose {
          rotation -0.5776567755050862 -0.5771969549958951 0.5771969549958951 2.0939354039397986
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0 0 1
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.01
                radius 0.025
              }
            }
          ]
        }
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    DEF RIGHT_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        position -20.216961239977245
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
        }
      ]
      endPoint DEF RIGHT_WHEEL Solid {
        translation 0.029999999999999805 0.025 0
        rotation -1 0 0 0.32842998874106355
        children [
          DEF WHEEL Pose {
            rotation -0.5776567755050862 -0.5771969549958951 0.5771969549958951 2.0939354039397986
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.196078 0.196078 0.196078
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    DEF DS0 DistanceSensor {
      translation 0.045 0.063 0
      children [
        DEF INFRARED Pose {
          rotation -0.5776567755050862 -0.5771969549958951 0.5771969549958951 2.0939354039397986
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds0"
      lookupTable [
        0 1000 0
        0.01 1000 0.01
        0.1 500 0.01
        0.9 50 0.01
        1 10 0
      ]
      aperture 1
    }
    DEF DS1 DistanceSensor {
      translation 0.0225 0.063 -0.039
      rotation 0 1 0 1.0472003061004251
      children [
        DEF INFRARED Pose {
          rotation -0.5776567755050862 -0.5771969549958951 0.5771969549958951 2.0939354039397986
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds1"
      lookupTable [
        0 1000 0
        0.01 1000 0.01
        0.1 500 0.01
        0.9 50 0.01
        1 10 0
      ]
      aperture 1
    }
    DEF DS6 DistanceSensor {
      translation 0 0.063 -0.044
      rotation 0 1 0 1.5707903061004251
      children [
        DEF INFRARED Pose {
          rotation -0.5776567755050862 -0.5771969549958951 0.5771969549958951 2.0939354039397986
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds6"
      lookupTable [
        0 1000 0
        0.01 1000 0.01
        0.1 500 0.01
        0.9 50 0.01
        1 10 0
      ]
      aperture 1
    }
    DEF DS4 DistanceSensor {
      translation 0.039 0.063 -0.0233
      rotation 0 -1 0 -0.5236053071795865
      children [
        DEF INFRARED Pose {
          rotation -0.5776567755050862 -0.5771969549958951 0.5771969549958951 2.0939354039397986
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds4"
      lookupTable [
        0 1000 0
        0.01 1000 0.01
        0.1 500 0.01
        0.9 50 0.01
        1 10 0
      ]
      aperture 1
    }
    DEF DS2 DistanceSensor {
      translation -0.0225 0.063 -0.039
      rotation 0 1 0 2.094399693899575
      children [
        DEF INFRARED Pose {
          rotation -0.5776567755050862 -0.5771969549958951 0.5771969549958951 2.0939354039397986
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds2"
      lookupTable [
        0 1000 0
        0.01 1000 0.01
        0.1 500 0.01
        0.9 50 0.01
        1 10 0
      ]
      aperture 1
    }
    DEF DS3 DistanceSensor {
      translation -0.045 0.063 0
      rotation 0 1 0 3.141592653589793
      children [
        DEF INFRARED Pose {
          rotation -0.5776567755050862 -0.5771969549958951 0.5771969549958951 2.0939354039397986
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds3"
      lookupTable [
        0 1000 0
        0.01 1000 0.01
        0.1 500 0.01
        0.9 50 0.01
        1 10 0
      ]
      aperture 1
    }
    DEF DS5 DistanceSensor {
      translation -0.039 0.063 -0.022
      rotation 0 -1 0 -2.6179953071795863
      children [
        DEF INFRARED Pose {
          rotation -0.5776567755050862 -0.5771969549958951 0.5771969549958951 2.0939354039397986
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds5"
      lookupTable [
        0 1000 0
        0.01 1000 0.01
        0.1 500 0.01
        0.9 50 0.01
        1 10 0
      ]
      aperture 1
    }
    DEF LED LED {
      translation 0 0.08 0
      children [
        Group {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                roughness 1
                metalness 0
              }
              geometry DEF LED_SHAPE Sphere {
                radius 0.01
              }
            }
            PointLight {
              attenuation 0 0 1
              color 0 0 0
              intensity 0.5
              location 0 0.03 0
              on FALSE
              castShadows TRUE
            }
          ]
        }
      ]
      color [
        1 0 0
        0 1 0
        0 0 1
        1 1 0
        0 1 1
        1 0 1
        1 0.647059 0
        0.627451 0.12549 0.941176
        0.545098 0.411765 0.0784314
      ]
    }
  ]
  name "ROBOT_RED"
  boundingObject DEF MYBOT_BOUNDING Pose {
    translation 0 0.0415 0
    rotation -1 0 0 1.5707963267948966
    children [
      DEF BODY Cylinder {
        height 0.08
        radius 0.045
      }
    ]
  }
  physics DEF MYBOT_PHYSICS Physics {
    density -1
    mass 0.5
  }
  controller "nmp_controller_python_3"
}
